### DISCUSSION TITLE
Cancelled Github actions fails PR checks

### DISCUSSION THREAD

**Discussion by Pustelto** [community/community#26127] [Actions]
<p>Hi,</p>
<p>I have setup Github actions for our company private repo to automatically deploy static site when new commits are pushed to master. To do this I use Filters for GitHub Actions. But when I create PR with new content filter action end with neutral status and the rest of actions (build and deploy) are cancelled which cause check to fail. And thus PR cannot be merged unless I disable the checks as mandatory in repo settings. </p>
<p>Is this supposed behaivior? Or is there some way how to avoid this? I couldn’t find any relevant information about this. Below is my my workflow file:</p>
<pre><code>workflow "Deploy to Firebase" {
  on = "push"
  resolves = ["Deploy"]
}

action "On master branch" {
  uses = "actions/bin/filter@master"
  args = "branch master"
}

action "Build" {
  needs = "On master branch"
  uses = "./.github/gatsby-build"
}

action "Deploy" {
  needs = "Build"
  uses = "webscopeio/firebase-deploy-github-action@master"
  secrets = ["FIREBASE_AUTH_TOKEN"]
  args = "--only hosting"
}
</code></pre>
<p>Thanks for advice.</p>

**Comment by mcolyer** [comment]
<p>Can you share more details about how you’ve configured protected branches for the repository (maybe a screenshot of the settings)? If it’s sensitive, we can move this to <a href="mailto:support@github.com">support@github.com</a>.</p>

**Comment by Pustelto** [comment]
<p>I have currently checks disabled but this is the checks status I get: </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/8598cbbd6d50424d13153517d5cc817afc3e265c.png" data-download-href="/uploads/short-url/j3QRLteAM69BkiTTeo6n8c0onIo.png?dl=1" title="Screenshot 2018-12-13 at 09.08.57.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/10606260/181095544-093f4945-da52-4b1c-98c1-383921ab1568.png" alt="Screenshot 2018-12-13 at 09.08.57.png" data-base62-sha1="j3QRLteAM69BkiTTeo6n8c0onIo" width="690" height="172" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/8/8598cbbd6d50424d13153517d5cc817afc3e265c_2_690x172.png, https://aws1.discourse-cdn.com/github/original/2X/8/8598cbbd6d50424d13153517d5cc817afc3e265c.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/8/8598cbbd6d50424d13153517d5cc817afc3e265c.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/8598cbbd6d50424d13153517d5cc817afc3e265c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2018-12-13 at 09.08.57.png</span><span class="informations">835×209 20.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Since the action is only build and deploy on master branch (no check for pushing to branch like test etc.)</p>

**Comment by jlelse** [comment]
<p>Is there any update on how you resolved this? I have the same problem. I created multiple workflows (one for every branch etc, one only for master and one only for develop). But when one of the workflows cancels (like when I push to develop, the workflow for master cancels), I get this red cross. Is there any possibility to ignore that and show a green check?</p>

**Comment by judoole** [comment]
<p>Does it change if you push the  <strong>Build</strong> -step one up? My thought was just that then the workflow would have at least one step that is green before filter cancels.</p>

**Comment by NiklasMerz** [comment]
<p>This results in an overall failed check for me, too.</p>

**Comment by NiklasMerz** [comment]
<p><a href="https://github.com/actions/bin/issues/13" rel="nofollow noopener">This issue</a> in the actions repo deals about that. too.</p>

**Comment by johnnyreilly** [comment]
<p>Me too - details here: <a href="https://github.com/TypeStrong/ts-loader/pull/893" rel="nofollow noopener">https://github.com/TypeStrong/ts-loader/pull/893</a></p>

**Comment by Pustelto** [comment]
<p>Didn’t find a way how to fix this. So for now I have to live with it like this. We have not set checks as mandatory in the repo so we can merge even with faild checks, but it is still annoying.</p>

**Comment by Pustelto** [comment]
<p>I did try different order of the actions in the flow but with no result. It always shows checks as failed. </p>

**Comment by mcolyer** [comment]
<p>Thanks for providing that additional information, so this behavior is expected but can see how it’s not ideal. We’re working on seeing how we can improve the situation.</p>

**Comment by mcolyer** [comment]
<p>That’s currently the best workaround.</p>

**Comment by NiklasMerz** [comment, marked as answer]
<p>I played around a lot and found out this works a lot more like I expected if you place the filter as a “need” for the resolving action.</p>
<p>I modified the example workflow:</p>
<pre><code>workflow "Deploy to Firebase" {
  on = "push"
  resolves = ["Deploy"]
}

action "On master branch" {
  uses = "actions/bin/filter@master"
  args = "branch master"
}

action "Build" {
  uses = "./.github/gatsby-build"
}

action "Deploy" {
  needs = ["Build", "On master branch"]
  uses = "webscopeio/firebase-deploy-github-action@master"
  secrets = ["FIREBASE_AUTH_TOKEN"]
  args = "--only hosting"
}
</code></pre>
<p>Now the PR checks are green and every action gets the status neutral.</p>
<p><span class="mention">@mcolyerI</span> this like it should be done? Maybe some notes in doc would help.</p>

**Comment by NiklasMerz** [comment]
<p>I think I found the right spot. The filter hast to be a requirement for the resolving action.</p>
<p>Check my answer in this thread <a href="https://github.community/t/cancelled-github-actions-fails-pr-checks/16030/13">https://github.community/t5/GitHub-API-Development-and/Cancelled-Github-actions-fails-PR-checks/m-p/17464/highlight/true#M639</a></p>

**Comment by Pustelto** [comment]
<p>I have tested your solution and it works so I accept it as a solution. Though donwside is that some steps are executed which may in the future expend limits for actions if there are one. Could be problematic for quite complicated workflow. Let’s hope GitHub folks will eventually fix this. </p>

**Comment by mcolyer** [comment]
<p>I think this works well - after reading more about this issue I think I originally misunderstood. We’re still looking to improve this area, so we’re taking your feedback to heart.</p>

**Comment by NiklasMerz** [comment]
<p>No I have almost the same problem again? Does anyone know how to handle this?</p>
<p><a href="https://github.community/t/github-actions-a-neutral-status-in-one-action-cancels-all-parallel-actions/16073">https://github.community/t5/GitHub-API-Development-and/GitHub-Actions-A-Neutral-Status-In-One-Action-Cancels-All/td-p/19189</a></p>

**Comment by jamesgorrie** [comment]
<p>Doesn’t this then only run the build on the master branch?</p>
<p>I am having the issue as I have the workflow of:</p>
<ul>
<li>
<p>Build</p>
</li>
<li>
<p>Test</p>
</li>
<li>
<p>Is master</p>
</li>
<li>
<p>Deploy</p>
</li>
</ul>
<p>In code:</p>
<pre><code>workflow "BTMD" {
  on = "push"
  resolves = ["Deploy"]
}

action "Build" {
  # blah
}

action "Test" {
  # blah
  needs = ["Build"]
}

action "Master" {
  # blah
  needs = ["Test"]
}

action "Deploy" {
  # blah
  needs = ["Master"]
}
</code></pre>
<p>The problem with this is when Master failes - the PR fails.</p>
<p>If we make Master and Test the needs of Deploy, if Master fails, Test doesn’t run., which we need it to to make sure the branch is good to merge…</p>
